### PR TITLE
 Add -config.cmake file to installation 

### PR DESCRIPTION
**Pull request by jparisu**
This PR adds the -config.cmake file to the project.
This file is very useful to link this project from others using find_package(cpp-peglib).
This find_package will populate ${cpp-peglib_INCLUDE_DIR} with the path to the include in install.

### CODE REVIEW

**Comment by jparisu** [comment]
Hi @yhirose 
Sorry for the delay in the response, I just did not have time to turn back to this propose.

The othe PR (#176) was a first draft, I hope this one is more compatible with your project.

`-config.cmake` file is not a requirement at all in cmake, but it is something useful to have and we would appreciate if you could take a look on it.
Please, feel free to add anything you want or to request any further change.

File: cmake/packaging/Config.cmake.in [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# This file is used by CMake in order to generate a -config.cmake installation file
+
+set(@PROJECT_NAME@_VERSION @PROJECT_VERSION@)
```
Comment (yhirose) [inline cmake/packaging/Config.cmake.in:3]: What does `@PROJECT_VERSION@` point to?

File: CMakeLists.txt [hunk 2]
```diff
@@ -17,10 +17,34 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
...
 enable_testing()
 
 install(FILES peglib.h DESTINATION include)
+
+##########
+# Generate -config.cmake file
+# This file will allow to link to this library (peglib.h) with cmake find_package
+
+# Install destination to -config.cmake
+set(INSTALL_DESTINATION_PATH ${LIB_INSTALL_DIR}/cmake/${PROJECT_NAME}${MSVCARCH_DIR_EXTENSION_EXT})
```
Comment (yhirose) [inline CMakeLists.txt:38]: `MSVCARCH_DIR_EXTENSION_EXT` looks Windows specific macro. Does this line work on other platforms like macOS and Linux?

File: CMakeLists.txt [hunk 3]
```diff
@@ -17,10 +17,34 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   set(add_link_deps Threads::Threads)
 endif()
 
+##############
+# Installation paths
+set(INCLUDE_INSTALL_DIR include/ CACHE PATH "Installation directory for C++ headers")
+set(LIB_INSTALL_DIR lib${LIB_SUFFIX}/ CACHE PATH "Installation directory for libraries")
+
```
Comment (yhirose) [inline CMakeLists.txt:24]: From line 20 through 23 should be moved after line 31 since nobody refers to the variables.

**Review by yhirose — COMMENTED** [review]
@jparisu, thanks for the update. I put some comments. Could you resolve these?

**Comment by yhirose** [comment]
I'll close it for now.

**Pull request closed without merging**